### ISSUE TITLE
Installing in Flatpak 

### ISSUE THREAD

**Issue by baarkerlounger** [Better-Player/espeakng-sys#7]
I'm trying to use this in flatpak but it fails to compile with:

```
error: linking with `cc` failed: exit status: 1
  |
  = note: LC_ALL="C" PATH="/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/bin:/app/bin:/usr/bin:/usr/lib/sdk/rust-stable/bin" VSLANG="1033" "cc" "-m64" "/tmp/rustcmhQ4Bt/symbols.o" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/build/espeakng-sys-a92592724e610fed/build_script_build-a92592724e610fed.build_script_build.748b7017790f522e-cgu.0.rcgu.o" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/build/espeakng-sys-a92592724e610fed/build_script_build-a92592724e610fed.49vsipmlusyfzxae.rcgu.o" "-Wl,--as-needed" "-L" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps" "-L" "/usr/lib/sdk/llvm16/lib" "-L" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/libbindgen-ac27aca53e58e6e7.rlib" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/libshlex-b2a8a911f5980326.rlib" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/librustc_hash-7b05e4fe75b58264.rlib" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/libregex-83aa7728f1ed84b5.rlib" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/libregex_automata-eb447f951ca62371.rlib" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/libaho_corasick-5758ce179aee613c.rlib" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/libregex_syntax-acc9cef604855d12.rlib" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/liblazycell-2c5b1cc05b5e4507.rlib" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/libpeeking_take_while-36ee259dee51fb21.rlib" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/libcexpr-7ea9e34a881791fa.rlib" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/libnom-7e6276d44588c7c9.rlib" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/libmemchr-333149cdd616a6c6.rlib" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/libclang_sys-3a5fedfa1ef6c87a.rlib" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/liblibc-073486283dd4d0b5.rlib" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/libglob-f70ac4a9ca01fc44.rlib" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/libquote-bf307f1af43a2537.rlib" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/libproc_macro2-239f331b1f056075.rlib" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/libunicode_ident-d135402c5721d012.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/libproc_macro-475723c0bac0f80c.rlib" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/liblazy_static-643a017f62beb017.rlib" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/deps/libbitflags-1926cec1a16924e0.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-8f88c761e33f2651.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-f9018f9cee1cc5ff.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/libobject-919f89587cbed68d.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/libmemchr-c6624cb4360306cd.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/libaddr2line-4930b3dc482158f7.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/libgimli-65bea4bb6af40828.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_demangle-bc6864da821ce9a2.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd_detect-1bccb7a942e1b311.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/libhashbrown-356231f77d1e268a.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_alloc-80ed5799bf463787.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/libminiz_oxide-9c3df673b2797081.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/libadler-c6afbee8d1102512.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-682387162b570769.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcfg_if-05a2cedbb78c1d4f.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-f002c8f83a289c4b.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-649be05783c8912e.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_core-13da980d6c74fec5.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-46a989d0e2cef827.rlib" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-b114db70ea0690b1.rlib" "-Wl,-Bdynamic" "-lclang" "-lc" "-lm" "-lrt" "-lpthread" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/usr/lib/sdk/rust-stable/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/home/dan/.cache/gnome-builder/projects/jogger/builds/xyz.slothlife.Jogger.json-flatpak-org.gnome.Platform-45beta-x86_64-espeak-ng/src/debug/build/espeakng-sys-a92592724e610fed/build_script_build-a92592724e610fed" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-nodefaultlibs"
  = note: /usr/lib/gcc/x86_64-unknown-linux-gnu/13.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /usr/lib/sdk/llvm16/lib/libclang.so: undefined reference to `LLVMInitializeWebAssemblyAsmPrinter@LLVM_16'
          /usr/lib/gcc/x86_64-unknown-linux-gnu/13.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /usr/lib/sdk/llvm16/lib/libclang.so: undefined reference to `LLVMInitializeWebAssemblyAsmParser@LLVM_16'
          /usr/lib/gcc/x86_64-unknown-linux-gnu/13.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /usr/lib/sdk/llvm16/lib/libclang.so: undefined reference to `LLVMInitializeWebAssemblyTarget@LLVM_16'
          /usr/lib/gcc/x86_64-unknown-linux-gnu/13.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /usr/lib/sdk/llvm16/lib/libclang.so: undefined reference to `LLVMInitializeWebAssemblyTargetInfo@LLVM_16'
          /usr/lib/gcc/x86_64-unknown-linux-gnu/13.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /usr/lib/sdk/llvm16/lib/libclang.so: undefined reference to `LLVMInitializeWebAssemblyTargetMC@LLVM_16'
          collect2: error: ld returned 1 exit status
          
  = note: some `extern` functions couldn't be found; some native libraries may need to be installed or have their path specified
  = note: use the `-l` flag to specify native libraries to link
  = note: use the `cargo:rustc-link-lib` directive to specify the native libraries to link with Cargo (see https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorustc-link-libkindname)

error: could not compile `espeakng-sys` (build script) due to previous error
warning: build failed, waiting for other jobs to finish...
```

`espeak-ng` is being installed with:

```json
{
            "name": "espeak-ng",
            "buildsystem" : "autotools",
            "no-parallel-make": true,
            "build-options": {
                "ldflags": "-Wl,--allow-multiple-definition"
            },
            "sources": [
                {
                    "type": "git",
                    "url": "https://github.com/espeak-ng/espeak-ng.git",
                    "tag": "1.51",
                    "commit": "2e9a5fccbb0095e87b2769e9249ea1f821918ecd"
                }
            ]
        },
```
and seems to install successfully.

**Comment by TobiasDeBruijn** [comment]
I'm not at all familiar with Flatpak unfortunately.
It looks like it's missing some symbols from the libclang library.
Could you try enabling the `clang-runtime` feature? I don't think it solves the issue but it's worth a shot.

**Comment by baarkerlounger** [comment]
Thank you, that works. Requires installing from git though - would you consider pushing a new release to crates.io for this?

**Comment by TobiasDeBruijn** [comment]
Oh sweet. I'll add that to the documentation as well.

I'll push a new version later today!

**Comment by TobiasDeBruijn** [comment]
Released as v0.1.3
